src/common/UsernameSet.cpp

```diff
@@ -43,15 +43,45 @@ std::set<QString>::size_type UsernameSet::size() const
 
 std::pair<UsernameSet::Iterator, bool> UsernameSet::insert(const QString &value)
 {
-    this->insertPrefix(value);
+    if (!value.isLower())
+    {
+        auto iter = this->items.find(value);
+        if (iter != this->items.end())
+        {
+            if (QString::compare(*iter, value, Qt::CaseSensitive) != 0)
+            {
+                this->items.erase(iter);
+            }
+            else
+            {
+                return {iter, false};
+            }
+        }
+    }
```

Comment:
Could we make this into an internal helper function at the top of the file? Since it's uused in the ::merge function

Reply:
You meant in another insert with move semantics? Sure